packages/clerk-js/src/utils/detect-ui-caller.ts

```diff
@@ -0,0 +1,49 @@
+function createStore(initialUsages: number) {
+  let state = initialUsages;
+
+  return {
+    get: () => state > 0,
+    increment: () => {
+      state++;
+    },
+    decrease: () => {
+      state = Math.max(state - 1, 0);
+    },
+  };
+}
+
+export const usageByUIComponents = createStore(0);
+
+const isThenable = (value: unknown): value is Promise<unknown> => {
```

Comment:
```suggestion
const isPromise = (value: unknown): value is Promise<unknown> => {
```

🙃 I'd suggest updating the name to match the type predicate as well, `thenable` was a bit hard to understand at a first glance